### PR TITLE
[MRG+2] tsdisplay

### LINKED ISSUE #261: tsdisplay equivalent
Whenever I see a cool feature in R that I want, my first stop is to check whether statsmodels supports it natively. If so, we should use their implementation. If not, we should add it. This is one of the cases where R's [`tsplot`](https://www.rdocumentation.org/packages/astsa/versions/1.9/topics/tsplot) is not implemented in statsmodels and it would be nice if we had an equivalent.

### PR DESCRIPTION
**Pull request by tgsmith61591**
# Description

Fixes #261, adding a new dataset and a utility to plot a time series and its relevant statistics. Also fixes a minor bug in BoxCox...

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation change

# How Has This Been Tested?

All existing tests pass, and the new example passes (which runs the function)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pmdarima/datasets/tests/test_load_datasets.py [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
     load_woolyrnq, load_ausbeer, load_austres,\
-    load_airpassengers, load_taylor, load_msft
+    load_airpassengers, load_taylor, load_msft, load_sunspots, _base as base
 
 import numpy as np
 import pandas as pd
 
 import pytest
 
+# todo: i don't code duplication, even if it's in tests.
```
Comment (aaronreidsmith) [inline pmdarima/datasets/tests/test_load_datasets.py]: > i don't code duplication

I don't _like_ code duplication?

**Comment by aaronreidsmith** [comment]
Seg fault on 3.5?

**Comment by aaronreidsmith** [comment]
> Seg fault on 3.5?

Re-running fixed it

File: pmdarima/datasets/tests/test_load_datasets.py [hunk 2]
```diff
@@ -37,3 +46,26 @@ def test_load(f):
...
+        # ensure shape is same for both
+        if n is None:
+            n = x.shape[0]
+        else:
+            assert x.shape[0] == n
+
+        if as_series:
+            assert isinstance(x, pd.Series)
+        else:
+            assert isinstance(x, np.ndarray)
```
Comment (charlesdrotar) [inline pmdarima/datasets/tests/test_load_datasets.py]: **Nit:** This block looks shared/duplicated between both functions.

File: pmdarima/datasets/_base.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+
```
[the remark this answers is not in the document]
Comment (tgsmith61591) [reply pmdarima/datasets/_base.py]: Meh. There's barely anything in that file. Doubt anyone is going to reach out with questions.

**Pull request merged**